Android/Self_Assessment_Devarshi/Intermediate/app/src/main/java/com/example/intermediate/SecondFragment.java

```diff
@@ -0,0 +1,175 @@
+package com.example.intermediate;
+
+import android.app.DatePickerDialog;
+import android.app.TimePickerDialog;
+import android.icu.text.SimpleDateFormat;
+import android.icu.util.Calendar;
+import android.os.Build;
+import android.os.Bundle;
+import android.text.InputType;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.DatePicker;
+import android.widget.EditText;
+import android.widget.Spinner;
+import android.widget.TimePicker;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.RequiresApi;
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentTransaction;
+import androidx.lifecycle.ViewModelProviders;
+import androidx.navigation.fragment.NavHostFragment;
+
+import com.example.intermediate.R;
+import com.example.intermediate.TransportViewModel;
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.firebase.firestore.DocumentReference;
+import com.google.firebase.firestore.FirebaseFirestore;
+
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+
+
+public class SecondFragment extends Fragment {
+
+    private TransportViewModel galleryViewModel;
```

Comment:
Please clean the code proper naming conventions must be a practice to follow and remove unused imports, methods, variables, print & debug statements. This applies globally.

Reply:
Done